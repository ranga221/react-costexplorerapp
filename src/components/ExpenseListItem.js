import React from 'react'
import { connect } from 'react-redux';
import { startRemoveExpense } from '../actions/expenses';
import { Link } from 'react-router-dom';
import numeral from 'numeral';

const ExpenseListItem = ({dispatch, id, description, amount, creadtedAt}) => (
<div>
<Link to={`/edit/${id}`}>
<h3>{description}</h3>

</Link>

<p>
{numeral(amount / 100).format('$0,0.00')}
-
{moment(creadtedAt).format('MMMM Do, YYYY')}
</p>
<button onClick={() => {
    dispatch(startRemoveExpense({ id }));
}}>Remove</button>
</div>

);

export default connect()(ExpenseListItem);