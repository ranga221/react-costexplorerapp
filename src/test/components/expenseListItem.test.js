import React from 'react';
import {shallow} from 'enzyme';
import ExpenseListItemForTest from '../../components/expenseListItemfortest';
import ExpenseListItem from '../../components/ExpenseListItem';

const expenses = [{
    id:1232,
    description:'Chew',
    note:'',
    amount: 2.00,
    createdAt:0 
}, {
    id:12325,
    description:'Gas Bill',
    note:'',
    amount: 3.00,
    createdAt:-1000 
},{
    id:12327,
    description:'Bus Bill',
    note:'',
    amount: 10.00,
    createdAt:-1000
}];

test('testing expense List item', () => {
const wrapper = shallow(<ExpenseListItemForTest {...expenses[0]} />);
expect(wrapper).toMatchSnapshot();
});