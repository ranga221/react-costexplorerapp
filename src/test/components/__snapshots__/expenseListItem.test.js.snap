// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing expense List item 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1232"
  >
    <h3>
      Chew
    </h3>
  </Link>
  <p>
    2
    -
  </p>
</div>
`;
